store/src/java/com/zimbra/cs/mailclient/MailConnection.java

```diff
@@ -90,14 +92,21 @@ public synchronized void connect() throws IOException {
                 }
                 break;
             }
-        } catch (IOException e) {
+        }
+        catch (SSLHandshakeException e) {
+            getLogger().debug(String.format("SSL Handshake Exception: %s", e.getMessage()));
```

Comment:
I think at debug level (which is off by default) it is safe and more useful to log the entire exception.